pydantic_settings/sources.py

```diff
@@ -664,6 +704,103 @@ def __repr__(self) -> str:
         )
 
 
+class ConfigFileSourceMixin(ABC):
+    def _read_files(self, files: PathType | None) -> dict[str, Any | None]:
```

Comment:
```suggestion
    def _read_files(self, files: PathType | None) -> dict[str, Any]:
```

Reply:
As in env settings, I think this is allowed in config files to have null values, as it's defined in json and in yaml specs ?